### PR TITLE
migate from setup.py to pyproject.toml & bump tox & declare support for python 3.12

### PR DESCRIPTION
**Pull request by Mogost**

### CODE REVIEW

**Review by sergei-iurchenko — APPROVED** [review]
Looks good for me. Good job!
Previous time a could not fix tox version issue but you did it!

**Comment by Mogost** [comment]
> Looks good for me. Good job! Previous time a could not fix tox version issue but you did it!

I have seen this and requested your review because of this.

File: tox.ini [hunk 1]
```diff
@@ -27,8 +27,8 @@ commands =
     unittest: coverage xml
     pytest: pytest --cov=. --ignore=.tox --disable-pytest-warnings --cov-report=xml --cov-append {toxinidir}
 setenv =
-    PYTHONDONTWRITEBYTECODE=1
```
Comment (sergei-iurchenko) [inline tox.ini:30]: may be leave PYTHONDONTWRITEBYTECODE=1 ?

Comment (Mogost) [reply tox.ini:30]: What this give us? As far I know the .pyc files will be ignored if the versions don't match.

File: tox.ini [hunk 2]
```diff
@@ -27,8 +27,8 @@ commands =
...
-    DJANGO_SETTINGS_MODULE=tests.settings
+    PYTHONPATH = {toxinidir}
```
Comment (Mogost) [inline tox.ini:30]:
```suggestion
    PYTHONPATH = {toxinidir}
    PYTHONDONTWRITEBYTECODE = 1
```

File: tox.ini (hunk 1 above)
Comment (Mogost) [reply tox.ini:30]: Ok. Fixed

**Pull request merged**